### PR TITLE
Add Design Guidelines Doc

### LINKED ISSUE #43: [Docs] : Add Design Guidelines Documentation
- [x] I have searched the [issues](https://github.com/jobtriage/jobtriage/issues) of this repository and believe that this is not a duplicate.

## Summary 💡

As the project grows, it is very difficult to follow the design rules, constraints, and principles. It is important to have design guidelines documentation to keep align everyone.

## Motivation 🔦

To make design and styles maintainable.

### PR DESCRIPTION
**Pull request by hiteshkundal**
# Pull Request Template

## Description
In DESIGN_GUIDELINES.md, we will keep adding our style and design rules, constraints, and principles.

Fixes #43 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: frontend/DESIGN_GUIDELINES.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# Design Guidelines for Job Triage
```
Comment (koushikmohan1996) [inline frontend/DESIGN_GUIDELINES.md]: Can we move this file somewhere common? Like `.github/contributing/`

Comment (hiteshkundal) [reply frontend/DESIGN_GUIDELINES.md]: We can move this to the root folder. I have seen this in material-ui project where they have common documentation at root location.

Comment (koushikmohan1996) [reply frontend/DESIGN_GUIDELINES.md]: Yes, looks good. Can you make the changes?

**Comment by koushikmohan1996** [comment]
Thanks :)

**Pull request merged**